okhttp/src/main/java/okhttp3/android/OkHttpAndroidClientFactory.kt

```diff
@@ -0,0 +1,58 @@
+package okhttp3.android
+
+import android.content.Context
+import okhttp3.OkHttpClient
+import okhttp3.internal.platform.Platform
+import okhttp3.internal.platform.android.BuildConfig
+import okhttp3.internal.tls.AllowlistedTrustManager
+
+class OkHttpAndroidClientFactory private constructor(val client: OkHttpClient) {
```

Comment:
Haven't given the Java API.  This is currently Kotlin first.  I expanded the scope to how do we hook into Android environment in one place, so we can understand things like the lifecycle and DEBUG mode etc.

This might be too far, or something to play with a bit, see what it's like via snapshots with real projects.